packages/rocketchat-api/server/api.js

```diff
@@ -161,14 +203,33 @@ class API extends Restivus {
 					});
 
 					logger.debug(`${ this.request.method.toUpperCase() }: ${ this.request.url }`);
+					const requestIp = this.request.headers['x-forwarded-for'] || this.request.connection.remoteAddress || this.request.socket.remoteAddress || this.request.connection.socket.remoteAddress;
+					const objectForRateLimitMatch = {
+						IPAddr: requestIp,
+						route: `${ this.request.route }${ this.request.method.toLowerCase() }`
+					};
 					let result;
 					try {
+						const shouldVerifyRateLimit = rateLimiterDictionary.hasOwnProperty(objectForRateLimitMatch.route);
```

Comment:
Can we please allow the `bot` role to bypass the rate limit? Bots will have a lot of issues with the API otherwise :)

Reply:
Yes @vynmera, the tests should also be out of the rate limiter. I'm changing, thanks for notifying.